packages/rx_bloc_cli/mason_templates/bricks/feature_profile/__brick__/lib/feature_profile/blocs/profile_bloc.dart

```diff
@@ -45,38 +44,26 @@ class ProfileBloc extends $ProfileBloc {
   @override
   Stream<bool> _mapToIsLoadingState() => loadingState;
 
-  @override
-  ConnectableStream<Result<bool>> _mapToSyncNotificationsStatusState() =>
+    @override
+  ConnectableStream<Result<bool>> _mapToAreNotificationsEnabledState() =>
       Rx.merge([
-        _$setNotificationsEvent.switchMap(
-          (subscribePushNotifications) {
-            if (subscribePushNotifications) {
-              return _notificationService
-                  .subscribe()
-                  .then((_) => true)
-                  .asResultStream(tag: tagNotificationSubscribe);
-            }
-
-            return _notificationService
-                .unsubscribe()
-                .then((_) => false)
-                .asResultStream(tag: tagNotificationUnsubscribe);
-          },
-        ),
+        _$toggleNotificationsEvent
+            .switchMap((_) => _notificationService
+                .areNotificationsEnabled()
+                .asStream()
+                .switchMap((areEnabled) => _notificationService
+                    .requestNotificationPermissions()
+                    .asStream()))
+            .switchMap((_) => _notificationService
+                .toggleNotifications()
```

Comment:
Do you think that we can request the notifications in the toggle notifications method instead in the bloc?

Reply:
Yes, done